cypress/integration/project/source_browsing.spec.ts

```diff
@@ -214,6 +214,22 @@ context("project source browsing", () => {
               .pickWithContent(["project-screen", "file-view"], "README.md")
               .should("exist");
 
+            // Going to a different path and then clicking on the project name
+            // sends us back to the project root, showing the README again.
+            commands.pick("project-screen", "commits-tab").click();
+            commands
+              .pick("horizontal-menu", "commits-tab")
+              .get("p")
+              .should("have.class", "active");
+            commands.pick("header", "entity-name").click();
+            commands
+              .pick("horizontal-menu", "files-tab")
+              .get("p")
+              .should("have.class", "active");
```

Comment:
Unfortunately this does not do what you would expect. `.get("p")` will select _all_ `p` elements and just find one that has the `active` class. In general we should also try to avoid using classes and element names to control the tests as they are prone to break.

My suggestion for this test would be to drop the assertion here since it is effectively checked by the assertion below that we are back at the project “home” screen. As for the assertion above I’d suggest to use `commands.pick("project-screen", "file-view").should("not.exist")`. We don’t want to check any specifics about the commits tab, just that we go back to the project home if we were not on it.